### PR TITLE
chore: Make compatible with .NET Framework 4.5 [DO NOT MERGE]

### PR DESCRIPTION
**Pull request by poppoerika**
- I was able to build `MomentoSigner` in Visual Studio 2022 on Windows Server 2012 R2 EC2 instance.
- Since xunit is not compatible with .NET Framework 4.5 (it's compatible with .NET Framework 4.5.2 or higher), I was not able to run `MomentoSignerTest`. https://xunit.net/docs/getting-started/netfx/visual-studio#multi-targeting
- Separately, we need to find out a way to add GitHub Action workflows that can be run on .NET Framework 4.5 (currently they only offer .NET Core). Maybe Kenny or other dotnet experts would know more about this?
- It appears that it's much easier/faster to develop/test this (.NET Framework 4.5) with non-M1 laptop or actual Windows PCs.

I might be missing things to watch out for this issue, so any advice/feedback would be greatly appreciated!

Ticket: https://github.com/momentohq/dev-eco-issue-tracker/issues/68

### CODE REVIEW

**Comment by danielamiao** [comment]
>Separately, we need to find out a way to add GitHub Action workflows that can be run on .NET Framework 4.5 (currently they only offer .NET Core). Maybe Kenny or other dotnet experts would know more about this?
@poppoerika could you just build this manually on your EC2 instance and just publish to nuget directly? I doubt we need to update this very much so continuous deployment via Github Actions is not a requirement.

**Comment by danielamiao** [comment]
@poppoerika also, beyond building could you try running the example app and make sure runtime is healthy? https://github.com/momentohq/client-sdk-examples/tree/main/dotnet/MomentoExamples/MomentoApplicationPresignedUrl

I'll DM you some instructions for how to run it

**Comment by poppoerika** [comment]
@wandaitzuchen @mjindalmotiv8 @virratanasangpunth 
Both Wanda and I verified that with this change in the SDK, we can run [this example](https://github.com/momentohq/client-sdk-examples/blob/main/dotnet/MomentoExamples/MomentoApplicationPresignedUrl/Program.cs) on Windows Server 2012 R2 and .NET Framework 4.5/4.5.2.
If this looks good, as @danielamiao mentioned, I can upload release builds to our jFrog public NuGet: https://momento.jfrog.io/ui/native/nuget-public/ manually.
Please take a look when you get a chance!

File: MomentoTest/MomentoSignerTest.cs [hunk 1]
```diff
@@ -1,99 +1,76 @@
 ﻿using System;
-using Xunit;
+using System.Text;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.IdentityModel.Logging;
```
Comment (wandaitzuchen) [inline MomentoTest/MomentoSignerTest.cs:5]: Don't think we need`Microsoft.IdentityModel.Logging`

File: MomentoTest/MomentoSignerTest.cs [hunk 2]
```diff
@@ -1,99 +1,76 @@
...
         }
 
-        [Fact]
-        public void TestPresignedUrlForSet()
+        static async Task Main(string[] args)
         {
-            MomentoSigner signer = new MomentoSigner(RS256_JWK);
-            uint expiryEpochSeconds = uint.MaxValue;
-            var req = new SigningRequest("testCacheName", "testCacheKey", CacheOperation.SET, expiryEpochSeconds)
+            IdentityModelEventSource.ShowPII = true;
```
Comment (wandaitzuchen) [inline MomentoTest/MomentoSignerTest.cs:36]: Can be removed?

File: MomentoTest/MomentoTest.csproj [hunk 3]
```diff
@@ -1,31 +1,15 @@
...
-</Project>
+<?xml version="1.0" encoding="utf-8"?>
+<Project Sdk="Microsoft.NET.Sdk">
+	<PropertyGroup>
+		<TargetFramework>net452</TargetFramework>
+		<IsPackable>false</IsPackable>
+		<OutputType>Exe</OutputType>
+	</PropertyGroup>
+	<ItemGroup>
+		<PackageReference Include="Microsoft.IdentityModel.Logging" Version="6.17.0" />
```
Comment (wandaitzuchen) [inline MomentoTest/MomentoTest.csproj:9]: Ditto

File: MomentoTest/MomentoSignerTest.cs (hunk 1 above)
Comment (poppoerika) [reply MomentoTest/MomentoSignerTest.cs:5]: Ah sorry I was using this to debug 🙇‍♀️ Will remove it.

**Review by wandaitzuchen — APPROVED** [review]

**Comment by mjindalmotiv8** [comment]
@poppoerika  @wandaitzuchen  can you please update https://www.notion.so/momentohq/Wyze-internal-onboarding-doc-f089cdf252e34a2f99b4d03ae465b010 with any additional instructions if there are any other things you had to do. Also once this get pushed to jfrog, can we please 

1) Verify that the project is able to pull down the SDK from jfrog
2) What is the way to pull down SDK updates? The current instructions use dotnet utility but I assume that dotnet utility is not available.

**Review by mjindalmotiv8 — DISMISSED** [review]

**Comment by poppoerika** [comment]
@mjindalmotiv8 @wandaitzuchen 

Manually published the `MomentoSdkDotnet45` package to our jFrog NuGet repo: https://momento.jfrog.io/ui/native/nuget-public/
I needed to update namespace so that it won't overlap with the existing SDK.

> can you please update https://www.notion.so/momentohq/Wyze-internal-onboarding-doc-f089cdf252e34a2f99b4d03ae465b010 with any additional instructions if there are any other things you had to do. Also once this get pushed to jfrog, can we please

Could you give me permission to edit that page?

> Verify that the project is able to pull down the SDK from jfrog

Verified that I can add the `MomentoSdkDotnet45` package to a project with .NET Framework 4.5 on Windows Server 2012 R2 following [this direction](https://github.com/momentohq/client-sdk-examples/tree/main/dotnet#cli-command-to-add-to-your-project).

> What is the way to pull down SDK updates? The current instructions use dotnet utility but I assume that dotnet utility is not available.

Users can just follow [this instructions](https://www.notion.so/momentohq/Wyze-CLI-Usage-Guide-438f0d672ca24187bf731bddf5c9bc11) on Windows as well to pull down the latest `MomentoSdkDotnet45` package.

**Comment by wandaitzuchen** [comment]
> Verified that I can add the MomentoSdkDotnet45 package to a project with .NET Framework 4.5 on Windows Server 2012 R2 following [this direction](https://github.com/momentohq/client-sdk-examples/tree/main/dotnet#cli-command-to-add-to-your-project).

The direction on the main dotnet example README does not apply to .NET Framework or user that does not have dotnet cli.  I added the README and example here: https://github.com/momentohq/client-sdk-examples/pull/84

> Users can just follow [this instructions](https://www.notion.so/momentohq/Wyze-CLI-Usage-Guide-438f0d672ca24187bf731bddf5c9bc11) on Windows as well to pull down the latest MomentoSdkDotnet45 package.

Wrong link?

File: MomentoTest/MomentoSignerTest.cs [hunk 4]
```diff
@@ -1,99 +1,74 @@
...
 
-        [Fact]
-        public void TestUrlEncoding()
-        {
-            MomentoSigner signer = new MomentoSigner(RS256_JWK);
-            uint expiryEpochSeconds = uint.MaxValue;
-            var testCacheKey = "#$&\\'+,/:;=?@[]";
-            var url = signer.CreatePresignedUrl("foobar.com", new SigningRequest("testCacheName", testCacheKey, CacheOperation.GET, expiryEpochSeconds));
+            // prepare requests
+            uint expiryEpochSeconds = (uint)DateTimeOffset.UtcNow.AddMinutes(URL_TTL_MINUTES).Subtract(new DateTime(1970, 1, 1)).TotalSeconds; ;
```
Comment (wandaitzuchen) [inline MomentoTest/MomentoSignerTest.cs:54]: extra `;` at the end

**Review by wandaitzuchen — APPROVED** [review]

**Pull request closed without merging**